src/js_native_api_v8.cc

```diff
@@ -3203,6 +3207,8 @@ napi_status NAPI_CDECL napi_get_typedarray_info(napi_env env,
       *type = napi_int32_array;
     } else if (value->IsUint32Array()) {
       *type = napi_uint32_array;
+    } else if (value->IsFloat16Array()) {
```

Comment:
Likewise here. Wrap the else if here in an `#ifdef NAPI_EXPERIMENTAL`

Reply:
Though I see the issue that previously we did not return any deterministic value at all. 
I guess there must be an "else" statement that returns some value like `std::numeric_limits<int32_t>::max()`.